pkg/maglev/maglev.go

```diff
@@ -60,16 +62,35 @@ func getOffsetAndSkip(backend string, m uint64) (uint64, uint64) {
 	return offset, skip
 }
 
-func getPermutation(backends []string, m uint64) []uint64 {
-	perm := make([]uint64, len(backends)*int(m))
+func getPermutation(backends []string, m uint64, numCPU int) []uint64 {
```

Comment:
A comment here to explain why this function takes in `numCPU` would be nice. Also just a brief comment to describe what the code is doing below (splitting work into chunks) and why this approach was taken over a semaphore, for example, would be nice to document in the code.

Reply:
I will push the comment once the CI has finished.